### DISCUSSION TITLE
option to toggle between "Playing" and "Listening to"

### DISCUSSION THREAD

**Discussion by prism2001** [s0hv/foo_discord_rich#14] [Ideas]
i have a last.fm plugin for Discord that allows you to set the status to "Listening to" rather than the usual "Playing" which i don't remember always being possible. i was wondering if this could potentially be a toggleable option in this plugin as well? would also be nice to have the option to change it to "music" rather than foobar2000 if we want to do so. 

here is what the last.fm rich presence looks like
![image](https://github.com/s0hv/foo_discord_rich/assets/11824514/d0d6a688-32b5-4d8f-8f33-fe60483ba8bc)
![image](https://github.com/s0hv/foo_discord_rich/assets/11824514/5f2ab57b-02a4-4b31-b1a0-50e955266b07)

**Comment by s0hv** [comment]
Could you link the last fm plugin you're using so I can take a look at it

**Reply by prism2001** [reply]
the last.fm plugin in question is actually part of a modded discord client. at first i thought it might be just a client-side thing that's only visible to users of said client, but i looked on mobile and can confirm it shows up globally, so i imagine it may be an option now that any rich presence plugin can apply. here's the plugin's code either way:

https://github.com/Vendicated/Vencord/blob/main/src/plugins/lastfm.tsx

**Reply by s0hv** [reply]
I suspected something like that. Since that is a modded client it uses different communication methods. This plugin has to rely on the rich presence sdk (known as game sdk nowadays), which does not support this feature.

You can see the documentation on the activity type in the [official game sdk documentation](https://discord.com/developers/docs/game-sdk/activities)

> ActivityType is strictly for the purpose of handling events that you receive from Discord; though the SDK/our API will not reject a payload with an ActivityType sent, it will be discarded and will not change anything in the client.
